webapp/tests/karma/ts/selectors/index.spec.ts

```diff
@@ -125,315 +125,336 @@ describe('Selectors', () => {
-      expect(Selectors.getCancelCallback(state)).to.deep.equal(clonedState.global.navigation.cancelCallback);
+      expect(Selectors.getCancelCallback.projector(state.global.navigation))
+        .to.deep.equal(clonedState.global.navigation.cancelCallback);
     });
 
     it('should getNavigation', () => {
-      expect(Selectors.getNavigation(state)).to.deep.equal(clonedState.global.navigation);
+      expect(Selectors.getNavigation.projector(state.global.navigation)).to.deep.equal(clonedState.global.navigation);
     });
 
     it('should getPreventNavigation', () => {
-      expect(Selectors.getPreventNavigation(state)).to.deep.equal(clonedState.global.navigation.preventNavigation);
+      expect(Selectors.getPreventNavigation.projector(state.global.navigation))
+        .to.deep.equal(clonedState.global.navigation.preventNavigation);
     });
 
     it('should getTitle', () => {
-      expect(Selectors.getTitle(state)).to.equal(clonedState.global.title);
+      expect(Selectors.getTitle.projector(state.global)).to.equal(clonedState.global.title);
     });
 
     it('should getPrivacyPolicyAccepted', () => {
-      expect(Selectors.getPrivacyPolicyAccepted(state)).to.equal(clonedState.global.privacyPolicyAccepted);
+      expect(Selectors.getPrivacyPolicyAccepted.projector(state.global))
+        .to.equal(clonedState.global.privacyPolicyAccepted);
     });
 
     it('should getShowPrivacyPolicy', () => {
-      expect(Selectors.getShowPrivacyPolicy(state)).to.equal(clonedState.global.showPrivacyPolicy);
+      expect(Selectors.getShowPrivacyPolicy.projector(state.global)).to.equal(clonedState.global.showPrivacyPolicy);
     });
 
     it('should getUnreadCount', () => {
-      expect(Selectors.getUnreadCount(state)).to.equal(clonedState.global.unreadCount);
+      expect(Selectors.getUnreadCount.projector(state.global)).to.equal(clonedState.global.unreadCount);
     });
 
     it('should getTranslationsLoaded', () => {
-      expect(Selectors.getTranslationsLoaded(state)).to.equal(clonedState.global.translationsLoaded);
+      expect(Selectors.getTranslationsLoaded.projector(state.global)).to.equal(clonedState.global.translationsLoaded);
     });
 
     it('should getUserFacilityId', () => {
-      expect(Selectors.getUserFacilityId(state)).to.equal(clonedState.global.userFacilityId);
+      expect(Selectors.getUserFacilityId.projector(state.global)).to.equal(clonedState.global.userFacilityId);
     });
 
     it('should getEnketoStatus', () => {
-      expect(Selectors.getEnketoStatus(state)).to.deep.equal(clonedState.global.enketoStatus);
+      expect(Selectors.getEnketoStatus.projector(state.global.enketoStatus))
+        .to.deep.equal(clonedState.global.enketoStatus);
     });
 
     it('should getEnketoEditedStatus', () => {
-      expect(Selectors.getEnketoEditedStatus(state)).to.equal(clonedState.global.enketoStatus.edited);
+      expect(Selectors.getEnketoEditedStatus.projector(state.global.enketoStatus))
```

Comment:
I'm wondering why it's necessary to directly serve the direct parent of the state property we want to get. Is there no way we can test these selectors by passing the whole state and making sure they're indeed picking the correct value? These tests seem devalued now that we spoon-feed the value they're supposed to return.

Reply:
Yes, agree, [this is](https://ngrx.io/guide/store/testing#testing-selectors) how they suggest to do it :(( 
I tried to find a way without changing too much...

It depends on how we define the selector, for example, the [getEnketoEditedStatus](https://github.com/medic/cht-core/blob/8e96ab03d51cf0b5c64632137ff0b6c417f1ec38/webapp/src/ts/selectors/index.ts#L41) can be created to always receive the full state: 
` getEnketoEditedStatus: createSelector((state) => state.global?.enketoStatus?.edited),`

Selectors will get more verbose

Reply:
How would selectors need to change in order for tests to stay as they are? An example would help, please don't go on to change _every_ selector :D

Reply:
I made a quick sample of selectors with unit tests. 

![Screen Shot 2023-01-12 at 9 12 48 pm](https://user-images.githubusercontent.com/66472237/212089753-379548f1-dc61-4b70-a36e-73fe91edaa48.png)

What do you think?

Reply:
Option 2 seems to have best test coverage and is pretty close to what we already have, right?

Reply:
Option 1 is also really good, a little more verbose.

Reply:
Okay, I'm going to update the selectors with Options # 2 - pass first level object of state. It's a good balance.

Reply:
Done